bftengine/src/bcstatetransfer/RangeValidationTree.hpp

```diff
@@ -0,0 +1,268 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the
+// LICENSE file.
+
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <memory>
+#include <unordered_map>
+#include <string>
+#include <cmath>
+#include <limits>
+
+#include <cryptopp/integer.h>
+
+#include "STDigest.hpp"
+#include "Serializable.h"
+#include "Logger.hpp"
+
+namespace bftEngine::bcst::impl {
+
+using RVBGroupId = uint64_t;
+using RVBId = uint64_t;
+using RVBIndex = uint64_t;
+
+// A Range Validation Tree (RVT)
+//
```

Comment:
please add a short comment about thread safety and about error handling:

1. no exception thrown
2. not thread safe
3. bad inputs values are asserted